File: fix_borders.js
const fs = require('fs');

const files = [
  'src/components/TestCaseRowNormal.tsx',
  'src/components/TestCaseRowTeaching.tsx',
  'src/components/TableFilters.tsx'
];

const replacements = [
  [/border-b-4 border-black/g, 'border-b'],
  [/border-r-4 border-black/g, 'border-r'],
  [/border-t-4 border-black/g, 'border-t'],
  [/border-l-4 border-black/g, 'border-l'],
  [/border-brutal/g, 'border rounded-lg'],
  [/shadow-brutal-lg/g, 'shadow-md'],
  [/shadow-brutal/g, 'shadow-sm'],
  [/hover:shadow-brutal/g, 'hover:shadow-md'],
  [/font-black/g, 'font-semibold'],
  [/border-2 border-black/g, 'border'],
];

files.forEach(filepath => {
  try {
    let content = fs.readFileSync(filepath, 'utf8');
    
    replacements.forEach(([pattern, replacement]) => {
      content = content.replace(pattern, replacement);
    });
    
    fs.writeFileSync(filepath, content, 'utf8');
    console.log(`Fixed ${filepath}`);
  } catch (error) {
    console.error(`Error fixing ${filepath}:`, error.message);
  }
});

console.log('Done!');

File: fix_borders.py
#!/usr/bin/env python3
import re

files_to_fix = [
    'src/components/TestCaseRowNormal.tsx',
    'src/components/TestCaseRowTeaching.tsx',
    'src/components/TableFilters.tsx'
]

replacements = [
    (r'border-b-4 border-black', 'border-b'),
    (r'border-r-4 border-black', 'border-r'),
    (r'border-t-4 border-black', 'border-t'),
    (r'border-l-4 border-black', 'border-l'),
    (r'border-brutal', 'border rounded-lg'),
    (r'shadow-brutal-lg', 'shadow-md'),
    (r'shadow-brutal', 'shadow-sm'),
    (r'hover:shadow-brutal', 'hover:shadow-md'),
    (r'font-black', 'font-semibold'),
    (r'border-2 border-black', 'border'),
]

for filepath in files_to_fix:
    try:
        with open(filepath, 'r') as f:
            content = f.read()
        
        for old, new in replacements:
            content = re.sub(old, new, content)
        
        with open(filepath, 'w') as f:
            f.write(content)
        
        print(f"Fixed {filepath}")
    except Exception as e:
        print(f"Error fixing {filepath}: {e}")

print("Done!")
